lib/internal/modules/cjs/loader.js

```diff
@@ -240,7 +241,21 @@ Module._findPath = function(request, paths, isMain) {
     var rc = stat(basePath);
     if (!trailingSlash) {
       if (rc === 0) {  // File.
-        if (preserveSymlinks && !isMain) {
+        if (!isMain) {
+          if (preserveSymlinks) {
+            filename = path.resolve(basePath);
+          } else {
+            filename = toRealPath(basePath);
+          }
+        } else if (preserveSymlinksMain) {
+          // for the main module, we use the preserveSymlinksMain flag instead
+          // mainly for backwards compatibility, as the preserveSymlinks flag
```

Comment:
Nit: `backwards` -> `backward`?

Reply:
other places in this file use "backwards" also, for what it's worth.  But I'm happy to change.